document-store/src/main/java/org/hypertrace/core/documentstore/expression/Filterable.java

```diff
@@ -0,0 +1,7 @@
+package org.hypertrace.core.documentstore.expression;
+
+/**
+ * A marker interface to represent that the expression is filterable and can be used in either the
+ * WHERE clause or the HAVING clause of the query.
```

Comment:
where and having would be different, I believe. Having would need an aggregate expression, while where would need to be non-aggregated (I really need a better term for that).

Reply:
While that's true, MongoDB allows alias names to be used in "HAVING" clauses. So, I think, it's better to leave it this way since an invalid expression would anyway fail while querying the DB.